app/models/dataset_extra.rb

```diff
@@ -162,6 +162,7 @@ def decorate(dataset)
     decorate_modaps_layers(dataset)
 
     dataset[:links] = Array.wrap(dataset[:links]) # Ensure links attribute is present
+    dataset[:browseable_collection] = dataset[:id] unless dataset[:links].select{|link| link[:rel].include? 'browse'}.empty?
```

Comment:
Style-wise, this line is a little confusing because it's a double-negative. I'd use positive logic:

    dataset[:browseable_collection] = dataset[:id] if dataset[:links].any? { |link| link[:rel].include?('browse') }